### ISSUE TITLE
Make tray icon themeable

### ISSUE THREAD

**Issue by andia89** [telegramdesktop/tdesktop#1908]
### Steps to reproduce
1. Open telegram on Linux
2. Enable tray icon
### Expected behaviour

Icon from generic place like `/usr/share/icons/hicolor` is used instead of a hardcoded one, so that different icon themes can provide tray icons for telegram instead of a hardcoded png. 
### Actual behaviour

An icon that is hardcoded in the binary is used, with absolutely no way to change apart from building telegramDesktop from source with changed png files...
### Configuration

Ubuntu 15.10 with appindicator installed

Telegram version 0.9.33

**Comment by auchri** [comment]
Shouldn't #1431 fix this issue?

**Comment by wa4557** [comment]
I don't think so because this is only for the app icon (the icon that shows up in the unity launcher for example), whereas the tray icon is still hardcoded.

If I made a pull request, is this something you would consider merging?

**Comment by auchri** [comment]
Duplicate #1199 

> If I made a pull request, is this something you would consider merging?

Yes, a PR would be great 👍

**Comment by john-preston** [comment]
What will happen with the coloured unread counters that are placed upon the icon (currently red or grey, depending on the muted status of the conversations with a positive unread counter)? They will be placed that way above an arbitrary tray icon?

**Comment by wa4557** [comment]
I suppose that's up to the icon theme to decide. If they  ship icons for all different states, they should show up

**Comment by john-preston** [comment]
@wa4557 I just don't understand (and afraid a bit) how this is going to look in the pull request — the counter is added uniformly there using internal styles.

**Comment by wa4557** [comment]
Is it? As far as I understand the code there are a lot of different png files for all different counter values. So for me there's not really a difference if this is done internally (this will be the fallback oc), or via an icon theme 

I could be wrong though, and misunderstand  the code...

**Comment by DavidVentura** [comment]
Any news on this? Plasma uses the theme icon but the status bar still shows the original blue icon

![teleg1](https://cloud.githubusercontent.com/assets/3650670/15986265/f3a01830-2fd9-11e6-8690-54ac71a42b37.png)
![teleg2](https://cloud.githubusercontent.com/assets/3650670/15986269/04ec1634-2fda-11e6-9474-71dd16164f6f.png)

**Comment by auchri** [comment]
@wa4557 Did you make some progress on this issue?

**Comment by wa4557** [comment]
Ah yeah I thought I commented on that. I coded around only to find out, that this wont work. The problem is that telegram doesn't use system Qt libraries, which consequentely means that the icon theme used is not recognized (I'm not sure how this is done by system Qt).

So this means that I was not able to use icons from `hicolor` or other icon themes

**Comment by DavidVentura** [comment]
Would you at least keep/distribute a directory named 'icons' with the icon? Also embed the default icon in the binary as it's done now in case the icon gets deleted. While this won't work with general themes, at least the interested users can switch the icon.

Maybe don't even ship the icon, but load it on launch if it's present.

**Comment by auchri** [comment]
> The problem is that telegram doesn't use system Qt libraries, which consequentely means that the icon theme used is not recognized (I'm not sure how this is done by system Qt).

But it works for the app icon .. ? #1431

**Comment by wa4557** [comment]
Well that's a different story. 

In order for telegram using system tray icons, I need the `QIcon::fromTheme` method, which as I said doesn't work with non system Qt libraries.

For the app icon it is as simple as changing a line in the `.desktop` file from `Icon=/path/to/a/hardcoded/icon.png` to a simple `Icon=telegram` the rest is done by the DE.

**Comment by DavidVentura** [comment]
Would it be possible to try and load an external file if it exists, and if it doesn't just fall back to the current icon (embedded in the binary)?

**Comment by wa4557** [comment]
@DavidVentura This is what I'm trying to do right now. We'll see if it works

**Comment by AndydeCleyre** [comment]
[This hack](https://github.com/telegramdesktop/tdesktop/issues/2230#issuecomment-239507699), modified for your color/icon preferences, may be useful for some of you until this is sorted.

**Comment by Aokromes** [comment]
#1199 ?

**Comment by mrjovanovic** [comment]
@Aokromes that is for the application icon, not the tray.

**Comment by Aokromes** [comment]
Dunno https://github.com/telegramdesktop/tdesktop/issues/1353#issuecomment-291024824 mentions that has tray.

**Comment by AndydeCleyre** [comment]
I had this themed with the following custom icons, and many symlinks for different message counts, in `~/.local/share/TelegramDesktop/tdata/ticons`:

```bash
ico_22_1.png
icomute_22_0.png
icomute_22_1.png
icon_22_0.png
icon_22_1.png
icon_22_2.png
icon_22_3.png
```

But it seems with the latest update from `1.5.4` to `1.5.8` the tray icon has reverted to the blinding blue out-of-place icon. Did the necessary location or names of the custom icons change?

**Comment by john-preston** [comment]
@AndydeCleyre The name changed from ico to icon.

**Comment by AndydeCleyre** [comment]
So after #7111, in order to choose the tray icon displayed on plasma desktop, the proper way becomes to have icons in the theme named `telegram-{,attention-,mute-}panel`?

**Comment by ilya-fedin** [comment]
+

**Comment by AndydeCleyre** [comment]
Is this in `1.9.12`?

![image](https://user-images.githubusercontent.com/1787385/74365812-cc892b80-4d9c-11ea-8784-d0e7f802d1cd.png)

![image](https://user-images.githubusercontent.com/1787385/74365912-fa6e7000-4d9c-11ea-95b0-50f012cd4389.png)

**Comment by ilya-fedin** [comment]
Yep

**Comment by AndydeCleyre** [comment]
How can I debug this, then?

**Comment by AndydeCleyre** [comment]
Looks like the regression is now being discussed at #7192.

**Comment by Rnksts** [comment]
No solution for Windows 10 users?

**Comment by ilya-fedin** [comment]
> No solution for Windows 10 users?

There is no such thing as "system icon theme" on Windows AFAIK

**Comment by Rnksts** [comment]
True but, after seeing #3301 and #1353 closed in favor of this issue, closed too, I wanted to remind or at least confirm if we are just out of luck.

**Comment by ilya-fedin** [comment]
> True but, after seeing #3301 and #1353 closed in favor of this issue, closed too

Since things changed, I think, you can open new issue
Or ask @Aokromes to reopen one of them (they have examples of how that can be done and discussion)

**Comment by AndydeCleyre** [comment]
Is this still supposed to work, or is there a new method? It seems broken again in `2.5.1` (see #6721).

**Comment by ilya-fedin** [comment]
It works, with counter, but works :)

**Comment by AndydeCleyre** [comment]
@ilya-fedin 

> It works, with counter, but works :)

Not entirely, anymore. Since `2.5.1`, aside from the addition of an unwanted and poorly colored counter, the rest of the icon now fails to color properly when that counter is shown, instead just turning black.

Attention icon before `2.5.1`:
![before](https://user-images.githubusercontent.com/1787385/103310007-953b6180-49e4-11eb-90ef-6a6a95f305e7.png)

Attention icon after `2.5.1`:
![after](https://user-images.githubusercontent.com/1787385/103309621-915b0f80-49e3-11eb-8d44-73544418814d.png)

So @ilya-fedin and @Aokromes, which is appropriate?
1. re-open this issue
2. treat the color regression issue as within scope of the `TDESKTOP_DISABLE_TRAY_COUNTER` regression issue (#6721)
3. open a new issue for the color regression alone

**Comment by ilya-fedin** [comment]
I don't think it is a regression. No one claimed that tdesktop supports KDE's color schemes (and most likely they won't be support due to the need to link to KDE libraries). You should use pre-colored icon theme.

**Comment by AndydeCleyre** [comment]
> most likely they won't be supported due to the need to link to KDE libraries

I think there may be a misunderstanding here. AFAIK no KDE libraries are currently linked, yet color schemes are indeed supported in the icon (fully before `2.5.1`, broken with `2.5.1` for *attention* icon but still working for *no-messages* icon).

All that's needed is to let plasma use the theme's svg, and it will do the coloring. Here's a sample svg that calls on the theme color (telegram doesn't have to be involved, really):

<details>
<summary>Expand</summary>

```css
<svg xmlns="http://www.w3.org/2000/svg" width="22" height="22" version="1">
 <defs>
  <style id="current-color-scheme" type="text/css">
   .ColorScheme-Text { color:#dfdfdf; } .ColorScheme-Highlight { color:#4285f4; }
  </style>
 </defs>
 <circle style="fill:currentColor" class="ColorScheme-Highlight" cx="6" cy="17" r="2"/>
 <path style="fill:currentColor" class="ColorScheme-Text" d="M 18.687469,18.303647 C 18.988656,17.713495 20.719303,7.6225537 20.966707,5.0140234 21.234358,2.4291929 19.838794,2.9098641 18.554652,3.3465233 15.404153,4.4215289 5.4432767,8.2785986 2.7167917,9.7167145 c -0.671236,0.28127 -0.961371,0.7585015 -0.472969,1.1715565 0.420514,0.350114 1.95926,0.830984 3.200218,1.075015 1.057861,0.323857 1.87956,0.01365 2.638057,-0.418712 1.822812,-1.092109 7.2543803,-4.0650435 7.5468763,-4.2586459 0.292496,-0.1936024 0.612628,0.1656936 0.448928,0.3396304 -0.163701,0.173936 -4.010423,3.6599335 -5.27339,5.2360675 -0.669101,0.866629 -0.649059,1.329311 0.0383,1.8758 1.54204,1.221933 5.662057,4.007627 6.275101,4.176004 0.781724,0.214705 1.239555,0.03683 1.569559,-0.609783 z"/>
</svg>
```
</details>

---

> #6721 is not a regression issue, it's a feature request from 201

Well it is a regression from my POV, as a feature was present and is now broken (`TDESKTOP_DISABLE_TRAY_COUNTER`), and the new broken state also introduces additional breakage (poor accessibility and aggressive anti-user coloring, as well as simply inconsistency -- not only between this icon and the rest of the system, but between this icon and the other telegram icon states).

**Comment by ilya-fedin** [comment]
> AFAIK no KDE libraries are currently linked, yet color schemes are indeed supported in the icon (fully before `2.5.1`, broken with `2.5.1` for _attention_ icon but still working for _no-messages_ icon).

It's because KDE colors when IconName is passed, but doesn't color IconPixmaps sadly.

> All that's needed is to let plasma use the theme's svg, and it will do the coloring.

We can't do that due to the counter.

> Well it is a regression from my POV, as a feature was present and is now broken (`TDESKTOP_DISABLE_TRAY_COUNTER`)

You can't call a feature something that was intended for temporary debug purposes. Can you complain if you used employees' entrance and someone closed it?

**Comment by AndydeCleyre** [comment]
You are right, I am coming from the perspective of folks who use the app. I am not describing the intentions of the developers which are sometimes at odds with accessibility needs, readability, and desktop consistency.

It sounds like your answer is WONTFIX, but it looks like my best bet is to open a separate issue, so I will do so.

**Comment by ilya-fedin** [comment]
> It sounds like your answer is WONTFIX

Most likely. Currently behavior is changed to not to add counter if "panel" icon is detected. But there are a high chance that as soon as it will land stable version, there will be complains that counter is lost. If this will happen, the only way is #6721, but new settings couldn't be added without @john-preston permission and he doesn't want to give it.

> looks like my best bet is to open a separate issue, so I will do so

It will be closed as a duplicate of #6721 most likely.
